nutils/function.py

```diff
@@ -4179,6 +4179,29 @@ class Namespace:
   >>> 'A_ij x_j' @ ns
   Array<3>
 
+  Sometimes the dimension of an expression cannot be determined, e.g. when
+  evaluating the identity array:
+
+  >>> ns.eval_ij('δ_ij')
+  Traceback (most recent call last):
+  ...
+  nutils.expression.ExpressionSyntaxError: Length of axis cannot be determined from the expression.
+  δ_ij
+    ^
+
+  There are two ways to inform the namespace of the correct lengths.  The first is to
+  assign fixed lengths to certain indices via the ``fixed_lengths`` argument:
+
+  >>> ns_fixed = function.Namespace(fixed_lengths=dict(i=2))
+  >>> ns_fixed.eval_ij('δ_ij')
+  Array<2,2>
+
+  The second is to define a fallback length via the ``fallback_length`` argument:
+
+  >>> ns_fallback = function.Namespace(fallback_length=2)
+  >>> ns_fallback.eval_ij('δ_ij')
+  Array<2,2>
+
```

Comment:
Just a thought: should we support the syntax ``fixed_lengths=dict(ijk=3, pqr=2)`` for ease of use? Since indices are always single character there are no ambiguities. We could unpack this in Namespace, I don't think there is value in passing it on to expression in this form.

Reply:
Done. Even shorter alternative: `Namespace(length_ijk=3, length_pqr=2)`.